util/test/computePerfStats

```diff
@@ -1,232 +1,296 @@
-      printf STATS "#%s ", $perfdate;
-      foreach my $key (@keys) {
-          print STATS "\t-";
-      }
-      print STATS " ### EXECUTION TIMED OUT ###\n";
-      exit 1;
-  }
-
-  printf STATS "%s ", $perfdate;
-  foreach my $key (@keys) {
-      print "Looking for $key...";
-      print STATS "\t";
-      my $found = 0;
-      foreach my $line (@file_lines) {
-          if ($line =~ m/\Q$key\E(\s*)(\S*)/ && $found == 0) {
-              print "found it: $2\n";
-              print STATS "$2";
-              $found = 1;
-          }
-      }
-      if ($found == 0) {
-          print STATS "-";
-          print "didn't find it\n";
-          $foundEverything = 0;
-
-      }
-  }
-  print STATS "\n";
-}
-
-if (($valid_output != 1 or $foundEverything != 1) and $exec_time_out eq "False") {
-    print "output was:\n";
-    # if a test output file is too long we clamp it down the first and last
-    # 1000 characters. Any non-printable characters are translated to '~' (for
-    # tests with binary output)
-    system(
-        "if [ \$(wc -c $test_output_file | awk \'{print \$1}\') -ge 2000 ]; then ".
-            "head -c 1000 $test_output_file | tr -C '[:alnum:][:space:][:punct:]' '~'; ".
-            "tail -c 1000 $test_output_file | tr -C '[:alnum:][:space:][:punct:]' '~'; ".
-        "else ".
-            "cat $test_output_file | tr -C '[:alnum:][:space:][:punct:]' '~'; ".
-        "fi");
-    # we can't be sure if we end on a new line when limiting by characters
-    print "\n";
-    exit 1;
-}
-
-# remove white space surrounding a string
-sub trim($) {
-    my $string = shift;
-    $string =~ s/^\s+//;
-    $string =~ s/\s+$//;
-    return $string;
-}
+# COMPUTE PERFORMANCE STATISTTICS
+# This is used by sub_test if start_test is called with the -performance flag.
+# 
+# For every performance run on a particular file, this script maintains a .dat
+# by searching the output of the run for values of performance keys specified
+# in .perfkeys. It then writes these values to the .dat for every run.
+
+from __future__ import print_function
+import argparse;
+import os;
+import time;
+import re;
+import sys;
+import difflib;
+
+args = None
+
+def main():
+    parser = parser_setup()
+    global args
+    args = parser.parse_args()
+
+    setup()
+    (keys, verify_keys) = read_key_file()
+    valid_output = validate_output(verify_keys)
+
+    found_everything = True
+    if (valid_output):
+        create_data_file(keys)
+        log_timeouts(keys)
+        found_everything = find_keys(keys)
+    
+    if not (valid_output and found_everything) and not args.exec_time_out:
+        cleanup()
+
+
+def find_keys(keys):
+    found_everything = True
+    with open(data_file, "a") as file:
+        file.write("%s " % args.perfdate)
+        for key in keys:
+            sys.stdout.write("Looking for %s..." % key)
+            file.write("\t")
+            found = False
+            regex = re.escape(key) + r"\s*(\S*)"
```

Comment:
I'm not sure if this is the right regex to be using.

I ran `start_test --performance test/` on one of the chap* boxes with no existing .dat files and got errors about missing keys. Might want to run that yourself (and make sure you empty `test/perfdat` to remove old .dat files)

After that I can run it with some of the existing .dat files in case there's any existing errors in them.